### PR TITLE
feat(prompts): make Document Insights guidance customizable

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary

- Make Document Insights analysis and presentation guidance editable through the existing Service Prompts settings in the WebUI and browser extension.
- Resolve one immutable authenticated-owner configuration per request and include the assembled prompt fingerprint in the existing response cache key. Prompt edits, resets, and in-flight changes cannot reuse results generated with different guidance.
- Preserve the exact default messages, fixed JSON instructions, requested-category/content carriers, provider configuration, content limits, and response normalization.
- Reuse the existing registry, storage, cache, and Settings editor instead of adding a separate configuration system.

Tracking: TASK-13198. Design: [Document Insights Service Prompt](Docs/Design/document-insights-service-prompt.md).

## Validation

- [x] Tests added/updated for behavior changes; initial regressions demonstrated ignored overrides and stale cache reuse.
- [x] Relevant backend tests: 111 passed, including 11 HTTP/storage/cache/model-boundary cases.
- [x] Shared Settings: 77 passed; WebUI Settings: 77 passed; prompt-service/transport: 124 passed.
- [x] Ruff lint/format, Python compilation, and diff checks pass.
- [x] Bandit: zero findings on both touched runtime files.
- [x] OpenAPI fingerprint unchanged; TypeScript schema generation passes.
- [x] Independent code review found no actionable issues.
- [x] Approved design and Backlog verification record included.

## UX Audit Checklist (v2 Stage 5)

- [x] WebUI/extension parity checked through shared Settings and WebUI-configured tests.
- Browser smoke, full-repository tests, and full frontend typecheck were not run. No new UI component or layout is introduced.
- Flashcards-specific checks are not applicable.

## Watchlists Accessibility Checklist (Group 09 Stage 5)

Not applicable: no Watchlists changes.

## Watchlists Scale Checklist (Group 10 Stage 5)

Not applicable: no Watchlists changes.

## Risk & Rollback

- Risk level: Medium. Custom guidance changes generated content; invalid saved guidance fails explicitly instead of returning cached/default results.
- Rollback: revert this PR. Existing cache entries expire normally; the additive setting requires no schema migration.

---
## Summary by cubic
Makes Document Insights analysis and presentation guidance editable through the existing Service Prompts settings in the WebUI and browser extension.

- Resolves one authenticated-owner prompt snapshot per request and includes its fingerprint in the existing response cache key, so prompt edits, resets, and in-flight changes cannot reuse results generated with different guidance.
- Prompt preparation lives in core; the endpoint supplies authenticated storage and maps resolution failures to HTTP with safe media/user/prompt identifiers in logs.
- Keeps the default messages, fixed JSON output instructions, requested-category carriers, provider configuration, content limits, and response normalization unchanged.
- Invalid saved guidance fails explicitly instead of falling back to cached or default results.
- Adds the `media.document.insights` definition to the existing registry and shared Settings editor; no new storage, cache, or configuration system.
- Rollback is a revert; existing cache entries expire normally and no schema migration is required.

<sup>Written for commit e409f6bd89a64dd3f816f42652ae84f2e8a34255. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_server/pull/2914?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tldw_Server_API/tests/Media/test_document_insights_service_prompt.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        state.reads.append(owner.id)
+        return state.databases[owner.id]
+
+    class MediaStore:
+        """Stand in for the existing media read boundary, keeping scope explicit."""
+
+        db_path_str = str(tmp_path / "media.db")
+
+        def get_media_by_id(self, media_id: int, **kwargs: Any) -> dict[str, Any]:
+            return {"id": media_id, "content": state.content, "type": "pdf"}
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/Media/test_document_insights_service_prompt.py:87]: Fixed in e409f6bd89. Added concise docstrings to every new fixture callback, fake method and test function. An AST audit reports no missing docstrings in the new test/core modules; all 12 feature integration tests pass.

File: tldw_Server_API/app/api/v1/endpoints/media/document_insights.py [hunk 2]
```diff
@@ -211,13 +189,39 @@ async def generate_document_insights(
...
+            try:
+                return resolve_service_prompt(prompts_db, "media.document.insights")
+            finally:
+                prompts_db.close_connection()
+
+        prompt = await asyncio.to_thread(resolve_and_close)
+    except HTTPException:
+        raise
+    except Exception as exc:
+        logger.error("Failed to resolve document insights guidance")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/app/api/v1/endpoints/media/document_insights.py]: Fixed in e409f6bd89. The prompt-resolution error includes safe media_id, user_id and the constant prompt_id without exception text, credentials, prompt content or database paths. Added a real-log regression that failed before the fix and now passes.

File: tldw_Server_API/app/api/v1/endpoints/media/document_insights.py [hunk 3]
```diff
@@ -211,13 +189,39 @@ async def generate_document_insights(
...
+        raise HTTPException(status_code=500, detail="Failed to load document insights guidance") from exc
+
+    system_prompt = (
+        f"{prompt.parts['analysis_guidance']}\n\nReturn JSON with this structure:\n"
+        '{"insights": [{"category": "...", "title": "...", "content": "..."}]}\n\n'
+        f"{prompt.parts['presentation_guidance']}\n- Return ONLY valid JSON, no other text\n"
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/app/api/v1/endpoints/media/document_insights.py]: Fixed in e409f6bd89. Prompt storage resolution, same-worker cleanup, fixed-instruction assembly and SHA-256 fingerprinting now live in core/Prompt_Management/document_insights.py. The endpoint supplies authenticated-owner storage and maps errors to HTTP. Existing HTTP/model/cache regressions pass unchanged; full focused backend suite: 112 passed.

File: tldw_Server_API/tests/Media/test_document_insights_service_prompt.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        kind="user",
+        user_id=state.owner,
+        username=f"owner-{state.owner}",
+        subject=f"user:{state.owner}",
+        token_type="access",
+    )
+    monkeypatch.setattr(endpoint, "get_prompts_db_for_user", database, raising=False)
+    monkeypatch.setattr(endpoint, "DEFAULT_LLM_PROVIDER", "openai")
+    monkeypatch.setattr(endpoint, "resolve_provider_api_key", lambda *a, **kw: ("test-key", None))
+    monkeypatch.setattr(endpoint, "load_and_log_configs", lambda: {"openai_api": {"model": "test-model"}})
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/Media/test_document_insights_service_prompt.py]: Fixed in e409f6bd89. Replaced the escaping dependency/provider/configuration and mid-request-edit lambdas with named local callbacks with parameter and return annotations. AST audit and Ruff pass; the 112-test focused backend suite remains green.

**Comment by rmusser01** [comment]
The repository owner explicitly approved merging PR #2914 and waiving the human-written Change summary requirement for this PR in the Codex task. All seven required checks pass on e409f6bd89a64dd3f816f42652ae84f2e8a34255, and all four Qodo review threads are resolved.
